lib/drivers/insuletDriver.js

```diff
@@ -614,18 +644,47 @@ module.exports = function (config) {
     });
   };
 
+  // these aren't history records, so we have to find them separately
+  var findPumpAlarmRecords = function (recordlist) {
+    var result = [];
+    for (var i = 0; i < recordlist.length; ++i) {
+      // TODO: check with KJQ about log_index of -1???
+      // doesn't have a timestamp, so I think it's not a valid record
+      if (recordlist[i].alarm != null && recordlist[i].log_index >= 0) {
+        result.push(i);
+      }
+    }
+    return result.sort(function compare(a, b) {
+      return recordlist[a].log_index - recordlist[b].log_index;
+    });
+  };
+
   var linkWizardRecords = function (data, bolusrecs) {
-    // we need to see if two boluses come from the same calculation (wizard) record
+    // we need to see if two (or more!) boluses come from the same calculation (wizard) record
     // if so, they're actually a dual bolus, and need to be consolidated
     // so we create a table of backlinks from the calc records to the bolus records that
     // refer to them
     var wizRecords = {};
     for (var b = 0; b < bolusrecs.length; ++b) {
-      var bolus = data.log_records[bolusrecs[b]];
-      var wiz_idx = bolusrecs[b] + bolus.detail.calculation_record_offset;
+      var bolus = data.log_records[bolusrecs[b]], wiz_idx;
```

Comment:
Whoa -- I'm not sure what you intended here, but that's probably not what you got. Or if it is, I don't understand why the ", wiz_idx" is on this instruction.